products/health-care/beneficiary-travel/engineering/spikes-for-api-team/api-versioning.md

```diff
@@ -0,0 +1,37 @@
+# Problem: API Versioning
+
+The API versioning schema should be related to the endpoints and functional not the release cycle
+
+## Risks
+
+Updating endpoints without updated functionality can cause unneeded overhead and maintenance for both client and API teams
```

Comment:
It can also introduce accidental security risks since using the latest/greatest versions of the endpoints is not automatic.

for example:

```mermaid
flowchart
  subgraph API
      direction TB
      v1 --> v2
  end
  subgraph releases
    direction TB
    v1.1 --adds endpoint--> v1
    v1.1 --> v1.2
    v1.2 --adds response attribute--> v1
    v1.2 --> v1.2.1
    v1.2.1 --patches security hole--> v1
    v1.2.1 --> v1.3
    v1.3 --removes endpoint--> v2   
  end  
```

If clients forget to update to `v1.2.1`, they don't receive the security fix. If versions are whole numbers, clients _will_ receive the security fix without addl work